src/java.desktop/share/classes/javax/swing/plaf/basic/BasicComboBoxUI.java

```diff
@@ -1118,35 +1118,33 @@ else if (comboBox.getModel().getSize() > 0) {
     // This is currently hacky...
     @Override
     public int getAccessibleChildrenCount(JComponent c) {
-        if ( comboBox.isEditable() ) {
-            return 2;
-        }
-        else {
-            return 1;
-        }
+        return 2;
     }
 
     // This is currently hacky...
     @Override
     public Accessible getAccessibleChild(JComponent c, int i) {
         // 0 = the popup
-        // 1 = the editor
+        // 1 = the editor for editable combobox and the arrow button for non-editable combobox
         switch ( i ) {
-        case 0:
-            if ( popup instanceof Accessible ) {
-                AccessibleContext ac = ((Accessible) popup).getAccessibleContext();
-                ac.setAccessibleParent(comboBox);
-                return(Accessible) popup;
-            }
-            break;
-        case 1:
-            if ( comboBox.isEditable()
-                 && (editor instanceof Accessible) ) {
-                AccessibleContext ac = ((Accessible) editor).getAccessibleContext();
-                ac.setAccessibleParent(comboBox);
-                return(Accessible) editor;
-            }
-            break;
+            case 0:
+                if ( popup instanceof Accessible accessiblePopup) {
+                    AccessibleContext ac = accessiblePopup.getAccessibleContext();
+                    ac.setAccessibleParent(comboBox);
```

Comment:
do we need to reset it back at some point?

Reply:
I don't think so, as comboBox is the accessible parent for popup. Moreover, this piece of code is unchanged for the current fix.